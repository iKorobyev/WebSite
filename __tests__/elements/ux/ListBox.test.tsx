import React from 'react'
import { cleanup, render, waitFor, act } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { ListBox } from '~ux'

// beforeAll(() => {})
// afterAll(() => {})
// beforeEach(() => {})
afterEach(() => {
  cleanup()
  jest.clearAllMocks()
})

describe('ListBox', () => {
  it('renders correctly', async () => {
    const props = {
      show: true,
      value: 'test',
      options: [{ label: 'Test', value: 'test' }],
      onChange: jest.fn(),
      onShow: jest.fn(),
      onHide: jest.fn()
    }

    const { container } = render(<ListBox {...props}>Test</ListBox>)
    await waitFor(() => container)

    expect(container).toMatchSnapshot()
  })
})
