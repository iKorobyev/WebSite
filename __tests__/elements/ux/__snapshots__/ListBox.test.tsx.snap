// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListBox renders correctly 1`] = `
<div>
  <div>
    <div
      class="relative mt-1"
    >
      <div
        class="relative w-full text-left bg-white rounded-lg shadow-md cursor-pointer focus:outline-none focus-visible:ring-2 focus-visible:ring-opacity-75 focus-visible:ring-white focus-visible:ring-offset-orange-300 focus-visible:ring-offset-2 focus-visible:border-indigo-500"
      >
        Test
      </div>
      <ul
        class="absolute w-full py-1 mt-1 overflow-auto bg-white rounded-md shadow-lg max-h-60 ring-1 ring-black ring-opacity-5 focus:outline-none z-1 transition ease-out duration-150 transform opacity-0 translate-y-5"
        id="headlessui-listbox-options-2"
        role="listbox"
        tabindex="0"
      >
        <li
          aria-selected="true"
          class="cursor-default select-none relative py-2 pl-10 pr-4 text-orange-900"
          id="headlessui-listbox-option-1"
          role="option"
          tabindex="-1"
        >
          <span
            class="block truncate font-medium"
          >
            Test
          </span>
          <span
            class="absolute inset-y-0 left-0 flex items-center pl-3"
          >
            <svg
              aria-hidden="true"
              class="w-10 h-10"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
